### PR TITLE
Add axiprop

### PR DESCRIPTION
**Pull request by ax3l**
Adds `axiprop`, a simple tool to compute optical propagation, based on the discrete Hankel and Fourier transforms.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] ~If static libraries are linked in, the license of the static library is packaged.~
- [x] ~Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).~
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by ax3l** [comment]
@conda-forge/help-python ready for review :)

File: recipes/axiprop/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{% set version = "0.5.2" %}
+
+package:
+  name: axiprop
+  version: {{ version }}
+
+source:
+  fn: {{ version }}.tar.gz
```
Comment (ocefpaf) [inline recipes/axiprop/meta.yaml:8]:
```suggestion
```

File: recipes/axiprop/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  url: https://github.com/hightower8083/axiprop/archive/refs/tags/{{ version }}.tar.gz
+  sha256: 60c35fa42b5d887aad6c241fc22fcd57df94e8ef8912de0a2f57443a8292b43c
+
+build:
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
```
Comment (ocefpaf) [inline recipes/axiprop/meta.yaml:15]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
```

File: recipes/axiprop/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
+    - setuptools
+  run:
+    - {{ pin_compatible('numpy') }}
```
Comment (ocefpaf) [inline recipes/axiprop/meta.yaml:23]: If you are not using numpy on host you don't need this.

Comment (ax3l) [reply recipes/axiprop/meta.yaml:23]:
```suggestion
    - numpy
```

File: recipes/axiprop/meta.yaml (hunk 2 above)
Comment (ax3l) [reply recipes/axiprop/meta.yaml:14]: Thanks! Waiting for resolution to https://github.com/conda-forge/staged-recipes/pull/24761#discussion_r1424676363 before applying :)

Comment (ax3l) [reply recipes/axiprop/meta.yaml:14]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv
```

**Pull request merged**